public/js/test_insight_socket.io.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+</head>
+
+<body>
+    <div id="updates"></div>
+    <script src="/js/vendor/socket.io.slim.js" type="application/javascript"></script>
+    <script type="application/javascript">
+        var updates = document.getElementById('updates')
+
+        var update = function (str) {
+            var pr = document.createElement('p')
+            pr.textContent = str
+            updates.appendChild(pr)
+        }
+
+        var sio = io(window.location.origin, { path: '/insight/socket.io', transports: ['websocket'], upgrade: false });
+
+        sio.on('connect', function () {
+            update('Connected. Updates will be shown below.')
+            sio.emit('subscribe', 'Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx');
```

Comment:
Joining a "room" on the server is accomplished by emitting a "subscribe" event. In this case the name of the room/subscription is the address.